projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	retries := 0
+	// This loop will run only if the number of reties doesn't exceed 3
+	for retries < 3 {
+		resp := getWeather()
+		// Close the response body after been fully read
+		defer resp.Body.Close()
+
+		// Read response body and store it in a var
+		body, err := io.ReadAll(resp.Body)
+		// Show error message if we cannot read the response body
+		if err != nil {
+			fmt.Print("Sorry we cannot get the weather!\n")
+			fmt.Fprintf(os.Stderr, "Failed to read response body: %v\n", err)
+			os.Exit(1)
+		}
+
+		// Handle cases depending on the Status code of the response
+		switch resp.StatusCode {
+		case 200:
+			// Set the retries number to 0 so we can run the program many time when the response is 200
+			retries = 0
+
+			// Convert response body from binary to string
+			sb := string(body)
+			fmt.Fprint(os.Stdout, sb+"\n")
+			os.Exit(0)
+		case 429:
+			handleRateLimited(resp.Header.Get("Retry-After"), retries)
```

Comment:
Can you add this logging and check that `retries` is working like you expect? If not, can you do some debugging as to why?

```suggestion
		case 429:
			handleRateLimited(resp.Header.Get("Retry-After"), retries)
			fmt.Fprintf(os.Stderr, "After rate-limited response, retries is: %d\n", retries)
```

Reply:
Yes , it wasn't working as expected ( the `retries` variable never get incremented) because the incrementation was inside a local scope. But we want to see the effect in the `main` function. Fixed by moving `retries++` to `main` just before closing the `for loop`